### PR TITLE
correct mariadb database and user handling

### PR DESCRIPTION
**Pull request by jochenmanz**
correct ensured users and database creation that i might have broken before 🙈 

config used for validation:

```
services.mysql.enable = true;
  services.mysql.package = pkgs.mariadb;
  services.mysql.initialDatabases = lib.mkDefault [{ name = "test_database"; }];
  services.mysql.ensureUsers = lib.mkDefault [
    {
      name = "test_user";
      password = "test_password";
      ensurePermissions = { "*.*" = "ALL PRIVILEGES"; };
    }
  ];
services.mysql.settings = {
    mysql = {
      user = "test";
      password = "test";
    };
    mysqldump = {
      user = "test";
      password = "test";
    };
    mysqladmin = {
      user = "test";
      password = "test";
    };
  };
```

### CODE REVIEW

File: src/modules/services/mysql.nix [hunk 1]
```diff
@@ -23,16 +27,16 @@ let
     PATH="${lib.makeBinPath [ cfg.package pkgs.coreutils ]}:$PATH"
     set -euo pipefail
 
-    while ! ${cfg.package}/bin/mysqladmin ping ${optionalString (!isMariaDB) "-u root --password=''"} --silent; do
+    while ! MYSQL_PWD="" ${cfg.package}/bin/mysqladmin ping  -u root --silent; do
```
Comment (jochenmanz) [inline src/modules/services/mysql.nix]: could we refactor this as a global (as in current shell) function so everyone can reuse this?

**Comment by domenkozar** [comment]
cc @shyim

**Review by shyim — APPROVED** [review]
tested works. the example config user was just wrong.

File: src/modules/services/mysql.nix (hunk 1 above)
Comment (shyim) [reply src/modules/services/mysql.nix]: I guess that won't work so easy. Maybe exposing it somehow with nix?

**Pull request merged**